src/NuGet.Core/NuGet.Packaging.Core/IPackageCoreWriter.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.Packaging.Core
+{
+    /// <summary>
+    /// Allows package modifications.
+    /// </summary>
+    public interface IPackageCoreWriter
+    {
+        /// <summary>
+        /// Remove if the path exists.
+        /// </summary>
+        /// <param name="path">Relative file path in package.</param>
+        /// <returns>True if the file existed before the remove.</returns>
```

Comment:
I actually got confused by this return text...

Reply:
i've removed the return value